File: mixin-padang-webface/src/main/js/padang/functions/dataset/ColumnExists.ts
/*
 * Copyright (c) 2020-2023 Datamixin.
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses/>.
 */
import FunctionPlan from "padang/plan/FunctionPlan";
import ParameterPlan from "padang/plan/ParameterPlan";
import ParameterPlanUtils from "padang/plan/ParameterPlanUtils";
import FunctionPlanRegistry from "padang/plan/FunctionPlanRegistry";

import DatasetFunction from "padang/functions/dataset/DatasetFunction";

export default class ColumnExists extends DatasetFunction {

    public static FUNCTION_NAME = "ColumnExists";

    constructor(
        public dataset: string,
        public name: string) {
        super(ColumnExists.FUNCTION_NAME, dataset);
    }

    public static NAME_PLAN: ParameterPlan = ParameterPlanUtils.createTextPlan(
        "name",
        "Name",
        "Column name subject to check"
    );

    public static getPlan(): FunctionPlan {
        let plan = new FunctionPlan(
            ColumnExists.FUNCTION_NAME,
            "Column Exists",
            "mdi-table-column",
            "Check a column is exists"
        );
        let parameters = plan.getParameterList();
        parameters.add(DatasetFunction.DATASET_PLAN);
        parameters.add(ColumnExists.NAME_PLAN);
        return plan;
    }
}

let registry = FunctionPlanRegistry.getInstance();
registry.registerPlan(ColumnExists.getPlan());

File: mixin-padang-garut/src/main/python/garut/functions/dataset/column_exists.py
#
# Copyright (c) 2020-2023 Datamixin.
#
# This program is free software: you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation, either version 3 of the License, or
# (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program. If not, see <http://www.gnu.org/licenses/>.
#
from typing import Dict
from garut.contexts.context import Context
from garut.instruction import InstructionFactory
from garut.functions.function import FunctionRegistry
from garut.functions.dataset.dataset_function import DatasetFunction


class ColumnExists(DatasetFunction):

    FUNCTION_NAME = "ColumnExists"

    NAME = "name"

    def execute(self, context: Context, options: Dict[str, any]) -> any:
        dataset = self.getDataset(options)
        column = options[ColumnExists.NAME]
        return column in dataset.columns.values


parameters = [DatasetFunction.DATASET, ColumnExists.NAME]
registry: FunctionRegistry = FunctionRegistry.getInstance()
registry.register(ColumnExists.FUNCTION_NAME, ColumnExists, parameters)

factory: InstructionFactory = InstructionFactory.getInstance()
factory.registerInstruction(ColumnExists.FUNCTION_NAME)
